main.cpp

```diff
@@ -19,13 +19,32 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+	CastCallBack(Rewriter& _rewriter) : rewriter_(_rewriter) {};
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
-    }
+	void run(const MatchFinder::MatchResult& Result) override {
+		if (auto* cast = Result.Nodes.getNodeAs<CStyleCastExpr>("cast")) {
+			auto type = Lexer::getSourceText(
+				CharSourceRange::getTokenRange(

```

Comment:
Please descrbe the difference between the use of `getCharRange()` and `getTokenRange()`, why `getTokenRange()` is used here.

Reply:
getCharRange() returns range from the starting to ending character by the specified locations.
getTokenRange() returns range specifying the start of the range and the start of the last token.
We may use any of them but if we don't specify arguments using getCharRange() will return following output (missing _t_ in _int_):
int main() {
    float f;
    int i = static_cast<in>(f);
    return 0;
}